### DISCUSSION TITLE
best possible recording

### DISCUSSION THREAD

**Discussion by inst5** [cables-gl/cables_docs#353] [Answers & Questions] [answered]
Hi,
may be this is a question asked a hunder times.

How can I export my animation in the best possible quality. 

For now I user MediaRecorder:
Mimetype_ video/webm; codecs=vp8
MBit: 30
FPS: 60

I convert the .webm to .mp4

But after working in AfterEffects and exporting my grient background are not smooth but I see segments.
But this is not visible in cable.gl

Thanks for help.

**Comment by marioepsley** [comment]
I'm on a new mac with the m1 max chip so quicktime seems to be the best local solution for me. 
Even better than that though is using an external video recorder. An Atomos ninja V or V+ both record at 4k up to 60fps. (spenny though) Means I can do a lot more processing on my computer and output silky smooth video at very high resolution.

With the mediaRecorder op I'm not to sure tbh. Would also be interested to find out.

**Comment by inst5** [comment]
Hi,
thx - ok, I see. Connecting a Ninja to capture is really simple and clever. - I even got on e here and will try this out.

How do mean that quicktime is a good local solution. Are you able to select .mov in the cables MediaRecorder?

**Reply by marioepsley** [reply]
I just use QuickTime as a quick screen recorder, there's no connection with cables. Anyway the quality seems good enough most of the time for me. Nice .mov :)

I guess the media recorder in cables will capture every printed frame? I've not tested it too much so not sure which settings are best. Ninjas my go to tbh.

**Comment by pandrr** [comment, marked as answer]
sadly we are very limited on what we can do in a browser
- media recorder output will always be compressed unless a browswee vendor decides to make uncompressed codecs available to js
- "downloading" images is slow and RAM consuming. it is not really possible in real time for longer sequences of images
- external recording software like quicktime or OBS is way more specialized but also requires a good gpu/cpu
- all content on cables.gl youtube channel is recorded using OBS, for shorter social media clips we use quicktime (the built in screen recorder of OSX)
